### PR TITLE
Fix `graphdef2mlir:const-values.pbtxt` test failure on s390x

### PR DESCRIPTION
**Pull request by kun-lu20**
Test case `//tensorflow/compiler/mlir/tensorflow/tests/graphdef2mlir:const-values.pbtxt.test` fails on s390x (big-endian arch) due to the endianness issue. The root cause of the failure is that the `tensorflow::LoadProtoFromBuffer()` function loads the `GraphDef` object directly from a Buffer which is in little-endian format (imported from file `const-values.pbtxt`): https://github.com/tensorflow/tensorflow/blob/7769d928bbf0751cbbb6fe4490747f1b12a2a47b/tensorflow/compiler/mlir/tensorflow/translate/tf_mlir_translate.cc#L55-L56, hence the tensor data in the `GraphDef` object is in LE format and will cause error on BE platforms.
 
This PR refactored the code in `tensorflow/core/util/tensor_bundle/byte_swap.cc` and added a helper function `ByteSwapTensorContentInGraphDef()` to do the byte swapping in this case. This function would be called in `tensorflow/compiler/mlir/tensorflow/translate/tf_mlir_translate.cc` after the invocation of `tensorflow::LoadProtoFromBuffer()` on BE platforms.
 
Besides fixing the above test failure, this PR can fix the following test failures from master branch on s390x as well and it won't cause any regressions on both s390x and other arches.
```bash
//tensorflow/compiler/mlir/lite/tests/end2end:fake_quant_per_channel.pbtxt.test
//tensorflow/compiler/mlir/lite/tests/end2end:fake_quant_per_channel_4bit.pbtxt.test
//tensorflow/compiler/mlir/lite/tests/end2end:fake_quant_without_identity.pbtxt.test
//tensorflow/compiler/mlir/lite/tests/end2end:fake_quant_without_identity_4bit.pbtxt.test
//tensorflow/python/distribute:cross_device_ops_test_2gpu
``` 

Signed-off-by: Kun-Lu <kun.lu@ibm.com>

### CODE REVIEW

**Comment by kun-lu20** [comment]
Hi @joker-eph ,

Could you please review this PR when you have some time? Thank you very much!

**Review by joker-eph — APPROVED** [review]

**Comment by joker-eph** [comment]
It's waiting on an internal reviewer to come back from vacations, should get done "soon".

**Comment by kun-lu20** [comment]
I see. Thank you so much @joker-eph

**Comment by kun-lu20** [comment]
Hi @joker-eph ,

Hope all is well with you.

Just want to check if there are any updates reg this PR from the internal review? 

Thank you very much!

**Comment by joker-eph** [comment]
I don't know what's going on here, but more importantly I'm no longer able to help!

Let me ping some folks who might be: @rohan100jain @rdzhabarov @tlongeri  ?

**Review by cantonios — APPROVED** [review]

**Comment by cantonios** [comment]
@kun-lu20 can you rebase please?  Thanks.

**Comment by kun-lu20** [comment]
Hi @cantonios ,

Thanks for your help! I've done the rebase, please take a look.

**Review by cantonios — APPROVED** [review]

**Comment by kun-lu20** [comment]
Hi @cantonios ,

I just checked the logs. I think the failures in `ARM CI / build (3.10)` and `MacOS CPU Python3` are irrelevant to this PR. 

In `import/copybara`, it shows "An error happened while migrating the change". Could you please help me check why this happened?

Thank you!

**Comment by cantonios** [comment]
The macos one looks like a failing test: `math_ops:approx_topk_test_cpu`.  Looks like this is currently failing at head, so not related to this change.  The arm CI looks like a bad pip setup. That's not blocking though, so it's okay.

**Comment by kun-lu20** [comment]
Thank you very much @cantonios !

**Comment by cantonios** [comment]
@kun-lu20 we're seeing a lot of build breakages.

```
tensorflow/compiler/mlir/tensorflow/translate/tf_mlir_translate.cc:41:10: error: module //tensorflow/compiler/mlir/tensorflow:translate_lib does not depend on a module exporting 'tensorflow/core/util/tensor_bundle/byte_swap_tensor.h'

...

In file included from third_party/tensorflow/core/util/tensor_bundle/tensor_bundle.cc:51:
tensorflow/core/util/tensor_bundle/byte_swap_tensor.h:37:36: error: unknown type name 'NodeDef'
Status ByteSwapTensorContentInNode(NodeDef& node);
                                   ^
1 error generated.
```

**Comment by kun-lu20** [comment]
Hi @cantonios ,

Thanks for letting me know. I'll look into this and get back to you soon.

**Comment by kun-lu20** [comment]
Hi @cantonios ,

Reg `error: unknown type name 'NodeDef'` in the error log, I've moved the `include` directive of `tensorflow/core/framework/node_def.pb.h` from `byte_swap_tensor.cc` into `byte_swap_tensor.h`. I think it will fix the above mentioned errors. The error didn't occur at my end.

Could you please review it again? Thank you very much!

File: tensorflow/compiler/mlir/tensorflow/translate/tf_mlir_translate.cc [hunk 1]
```diff
@@ -38,6 +38,7 @@ limitations under the License.
 #include "tensorflow/core/platform/errors.h"
 #include "tensorflow/core/platform/protobuf.h"
 #include "tensorflow/core/protobuf/graph_debug_info.pb.h"
+#include "tensorflow/core/util/tensor_bundle/byte_swap_tensor.h"
```
Comment (cantonios) [inline tensorflow/compiler/mlir/tensorflow/translate/tf_mlir_translate.cc:41]: the BUILD entry for this target needs to depend directly on the build target that exports the `byte_swap_tensor.h` header.

**Review by cantonios — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/mlir/tensorflow/translate/tf_mlir_translate.cc (hunk 1 above)
Comment (kun-lu20) [reply tensorflow/compiler/mlir/tensorflow/translate/tf_mlir_translate.cc:41]: Hi @cantonios ,

Thanks for the clarification! I've added target `//tensorflow/core/util/tensor_bundle:byteswaptensor` in the dep list of the target `translate_lib` in its BUILD entry. Could you please take a look?

**Review by cantonios — APPROVED** [review]

**Comment by kun-lu20** [comment]
Hi @cantonios ,

I found the `Windows Bazel` test showed `Internal CI infrastructure error`. I think it's not related to this code change, right?

Thank you!

**Comment by cantonios** [comment]
> Hi @cantonios ,
> 
> I found the `Windows Bazel` test showed `Internal CI infrastructure error`. I think it's not related to this code change, right?
> 
> Thank you!

I don't think so - it looks like it was manually aborted.  We're still waiting for other reviewers internally to approve.

**Comment by kun-lu20** [comment]
Thanks @cantonios !

**Pull request merged**